File: app/controllers/People.java
package controllers;

import models.*;
import org.codehaus.jackson.JsonNode;
import play.libs.Json;
import play.mvc.Controller;
import play.mvc.Result;
import service.post.Redis;

import java.util.Date;
import java.util.List;

public class People extends Controller {

    public static Result follow(){
        Date now = new Date();
        String follow_person_id = request().getQueryString("follow_person_id");
        String circle_id = request().getQueryString("circle_id");
        User connectedUser = Application.getLocalUser(session());
        User person = User.find.byId(Long.parseLong(follow_person_id));
        TContact contact = TContact.add(connectedUser,person,now);
        TCircle.addContactById(Long.parseLong(circle_id),contact,now);
        return ok("successed");
    }

    public static Result unfollow(){
        User connectedUser = Application.getLocalUser(session());
        String follow_person_id = request().getQueryString("follow_person_id");
        User person = User.find.byId(Long.parseLong(follow_person_id));
        TContact contact = TContact.findContact(connectedUser,person);
        contact.delete();
        return ok("successed");
    }


}

File: app/controllers/People.scala
package controllers

import play.api.mvc.{Action, Controller}
import models.{Contact, User}
import com.mongodb.casbah.commons.MongoDBObject
import java.util.Date
import org.bson.types.ObjectId
import securesocial.core._
import play.Logger
import org.joda.time.DateTime

object People extends Controller with SecureSocial{

  def avatar = SecuredAction { request =>
    implicit val user = request.user
    val personId = request.getQueryString("id").getOrElse("")
    val followed = Contact.findPersonByOwner(personId).isDefined;
    val person = User.findOneByStringId(personId).get
    val myself = user.identityId==person.id
    Ok(views.html.post.avatarModal(person,followed,myself))
  }

  def follow = SecuredAction { request =>
    implicit val user = request.user
    val now = DateTime.now()
    val personId = request.getQueryString("id").getOrElse("")
    val person = User.findOneByStringId(personId).get.id
    Contact.findPersonByOwner(personId) match {
      case None => Contact.save(Contact(owner = user.identityId,person = person,create_at = now))
      case _ =>
    }
    Ok
  }

  def unfollow = SecuredAction { request =>
    implicit val user = request.user
    val personId = request.getQueryString("id").getOrElse("")
    Contact.findPersonByOwner(personId) map { contact =>
      Contact.remove(contact)
    }
    Ok
  }

}
